src/apps/replication/client_ddl_tool/main.cpp

```diff
@@ -0,0 +1,102 @@
+#include <dsn/service_api_c.h>
+#include <dsn/ports.h>
+#include <dsn/dist/replication/empty_app.h>
+#include <dsn/dist/replication/client_ddl.h>
+#include <iostream>
+
+using namespace dsn::client;
+
+void usage(char* exe)
+{
+    std::cout << "Usage:" << std::endl;
+    std::cout << "\t" << exe << " <config.ini> create_app -name <app_name> -type <app_type> [-pc partition_count] [-rc replication_count]" << std::endl;
+    std::cout << "\t" << exe << " <config.ini> drop_app -name <app_name>" << std::endl;
+    std::cout << "\t\tpartition count must be a power of 2" << std::endl;
+    std::cout << "\t\tapp_name and app_type shoud be composed of a-z, 0-9 and underscore" << std::endl;
+    exit(-1);
+}
+
+extern void dsn_core_init();
+
+int init_environment(char* exe, char* config_file)
+{
+    // register all possible services
+    dsn::register_app<dsn::client::empty_app>("empty_app");
+
+    dsn_core_init();
+
+    //use config file to run
+    char arg1[] = "-app_list";
+    char arg2[] = "empty_app@1`";
+    char* argv[] = { exe, config_file, arg1, arg2};
+
+    dsn_run(4, argv, false);
+    dsn_mimic_app("empty_app", 1);
```

Comment:
we don't need empty_app anymore, just set [core] enable_default_app_mimic = true